Prevent Ansible args from leaking to create and destroy

tadeboro opened the pull request:
When we were removing the dependency on the sh Python package from the codebase, we inadvertently changed the behavior of the Ansible arguments that user supplies either on the command line on in the provisioner part of the scenario configuration (commit 1cefe5c21e8ef338a5769d65e4e0e3eb6b2256fc).

Changes in this commit restore the previous state when Ansible args were not passed to the create and destroy playbook executions.

Fixes #2963 

#### PR Type

- Bugfix Pull Request

ssbarnea commented:
I knew it! I deliberately simplified the code because I did not see why these should not be passed and because it did not break any tests.... so blame me for introducing that bug.

I approved the fix but I will let you decide if you want to merge it like this or if you can also add two bits to it:
* a small comment before the if explaining why we do not want args for create/destroy - why they are special
* a test that would fail if someone removes this code (if ansible args do slip in, again)

tadeboro commented:
>     * a small comment before the if explaining why we do not want args for create/destroy - why they are special

I will add this comment to the code.

>     * a test that would fail if someone removes this code (if ansible args do slip in, again)

The test is already there (https://github.com/ansible-community/molecule/blob/94624bcd5f119f5553b7ab0edd3855307bae01f4/lib/molecule/test/unit/provisioner/test_ansible_playbook.py#L109), but it failed to detect the change because we updated it by mistake. I already fixed the test, so we should be safe in the future (until we mess up the tests again, that is ;)).
